execution_workflows/DaPars2/Dapars2_snakemake/workflow/Snakefile

```diff
@@ -0,0 +1,333 @@
+"""
+
+import pandas as pd
+import os
+import sys
+
+configfile:"config/config.DaPars2.yaml"
+
+samples = pd.read_csv(config["samples"]).set_index("sample", drop=False)
+
+
+#-------------------------------------------------------------------------------
+localrules: finish
+
+
+rule finish:
+    """Rule that specifies the final output.
+    """
+    input:
+        final = expand(
+            os.path.join(config["out_dir"], "merge_{sample}"),
+            sample=samples.index.values)
+        # OUT1 = os.path.join(config["out_dir"],  "DaPars2_result_chr")
+        # os.path.join(config["out_dir"], "01_DaPars2.BED")
+
+# https://evodify.com/snakemake-checkpoint-tutorial/
+#-------------------------------------------------------------------------------
+# Preprocessing: obtain suitable input formats
+rule addChrToGtf:
+    """This step adds 'chr' prefix to gtf file.
+    """
+    input:
+        gtf=config['gtf']
+    output:
+        out=os.path.join(config["out_dir"], "annotation_Chr.gtf")
+    log:
+        os.path.join(config["local_log"], "addChrToGtf.log")
+    shell:
+        """(awk '{{if($1 !~ /^#/){{print "chr"$0}}else{{print $0}}}}' {input.gtf} > {output.out}) &> {log}"""
+
+
+rule addChrToGff3:
+    """This step adds 'chr' prefix to gtf file.
+    """
+    input:
+        gff3=config["gff3"]
+    output:
+        out=os.path.join(config["out_dir"], "annotation_Chr.gff3")
+    log:
+        os.path.join(config["local_log"], "addChrToGff3.log")
+    shell:
+        """(awk '{{if($1 !~ /^#/){{print "chr"$0}}else{{print $0}}}}' {input.gff3} > {output.out}) &> {log}"""
+
+
+rule getBed12_1:
+    """This step generates genePred from gff3. This is the first step for generate 12 column bed file.
+    """
+    input:
+        gff3=os.path.join(config["out_dir"], "annotation_Chr.gff3")
+    output:
+        out=os.path.join(config["out_dir"], "annotation_Chr.genePred")
+    conda:
+        os.path.join(config["envs"], "DaPars2.yaml")
+    container:
+        "docker://sambrycesmith/apaeval_dapars2:23d89d1_and_preprocessing"
+    log:
+        os.path.join(config["local_log"], "getBed12_1.log")
+    shell:
+        """(gff3ToGenePred {input.gff3} {output.out}) &> {log}"""
+
+
+rule getBed12_2:
+    """This step generates 12 column bed file from genePed. This is the second step for generate 12 column bed file
+    """
+    input:
+        genePred=os.path.join(config["out_dir"], "annotation_Chr.genePred")
+    output:
+        out=os.path.join(config["out_dir"], "annotation.bed")
+    conda:
+        os.path.join(config["envs"], "DaPars2.yaml")
+    container:
+        "docker://sambrycesmith/apaeval_dapars2:23d89d1_and_preprocessing"
+    log:
+        os.path.join(config["local_log"], "getBed12_2.log")
+    shell:
+        """(genePredToBed {input.genePred} {output.out}) &> {log}"""
+
+
+rule extractGeneIdAndName:
+    """This step extracts gene ID and gene symbol from gtf.
+    """
+    input:
+        gtf=config["gtf"]
```

Comment:
I guess this could also be done with the GFF3 annotation, but possibly needs adjustment of the python script. I think we can leave it as it is but maybe simplify things in the future.